src/plugins/visualize/public/application/utils/get_top_nav_config.tsx

```diff
@@ -245,6 +256,117 @@ export const getTopNavConfig = (
+              'visualize.topNavMenu.duplicateVisualizationButtonAriaLabel',
+              {
+                defaultMessage: 'Duplicate Visualization',
+              }
+            ),
+            testId: 'visualizeDuplicateButton',
+            disableButton: hasUnappliedChanges,
+            tooltip() {
+              if (hasUnappliedChanges) {
+                return i18n.translate(
+                  'visualize.topNavMenu.duplicateVisualizationDisabledButtonTooltip',
+                  {
+                    defaultMessage: 'Apply or Discard your changes before duplicating',
+                  }
+                );
+              }
+            },
+            run: (anchorElement: HTMLElement) => {
+              const getDuplicateWorkspaces = async (): Promise<WorkspaceAttribute[]> => {
+                let result;
+                try {
+                  result = await getWorkspacesWithWritePermission(http);
+                } catch (error) {
+                  notifications?.toasts.addDanger({
+                    title: i18n.translate(
+                      'savedObjectsManagement.objectsTable.duplicateWorkspaces.dangerNotification',
+                      {
+                        defaultMessage: 'Unable to get workspaces with write permission',
+                      }
+                    ),
+                    text: error instanceof Error ? error.message : JSON.stringify(error),
+                  });
+                }
+                if (result?.success) {
+                  return result.result?.workspaces ?? [];
+                } else {
+                  return [];
+                }
+              };
+
+              const onDuplicate = async (
+                visualizationSavedObjects: SavedObjectWithMetadata[],
+                includeReferencesDeep: boolean,
+                targetWorkspace: string
+              ) => {
+                // object must be visualization type
+                const objectsToDuplicate = visualizationSavedObjects.map((obj) => ({
+                  id: obj.id,
+                  type: 'visualization',
+                }));
+
+                try {
+                  await duplicateSavedObjects(
+                    http,
+                    objectsToDuplicate,
+                    includeReferencesDeep,
+                    targetWorkspace
+                  );
+                } catch (e) {
+                  notifications.toasts.addDanger({
+                    title: i18n.translate('visualize.topNavMenu.duplicate.dangerNotification', {
+                      defaultMessage: 'Unable to duplicate visualization',
+                    }),
+                  });
+                  throw e;
+                }
+                notifications.toasts.addSuccess({
+                  title: i18n.translate('visualize.topNavMenu.duplicate.successNotification', {
+                    defaultMessage: 'Duplicate visualization successfully',
+                  }),
+                });
```

Comment:
```suggestion
                try {
                  await duplicateSavedObjects(
                    http,
                    objectsToDuplicate,
                    includeReferencesDeep,
                    targetWorkspace
                  );
                  notifications.toasts.addSuccess({
                    title: i18n.translate('visualize.topNavMenu.duplicate.successNotification', {
                      defaultMessage: 'Duplicate visualization successfully',
                    }),
                  });
                } catch (e) {
                  notifications.toasts.addDanger({
                    title: i18n.translate('visualize.topNavMenu.duplicate.dangerNotification', {
                      defaultMessage: 'Unable to duplicate visualization',
                    }),
                  });
                }
```
I suggest to move success toast to right after `duplicateSavedObjects()` function call, and remove `throw e` in catch clause.